ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts

```diff
@@ -289,9 +289,12 @@ class DataGridComponent extends Component {
      * @internal
      */
     public override getOptions(): Partial<Options> {
+        const optionsCopy = merge(
+            this.options,
+            { gridOptions: this.grid?.getOptions(true) }
+        );
```

Comment:
I would completely replace `this.options.gridOptions` with the real grid user options, so maybe this will be better? Unfortunately, we still have to deal with the fact that in Dashboards we do not separate `userOptions` from `options`, and in this case `userOptions` are definitely more important.
```ts
const optionsCopy = merge(this.options);
optionsCopy.gridOptions = this.grid?.getOptions(true);
```